internal/provisioner/plain/pkg/git/git.go

```diff
@@ -0,0 +1,78 @@
+package git
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/operator-framework/rukpak/api/v1alpha1"
+)
+
+const (
+	defaultDirectory = "./manifests"
+	defaultBranch    = "main"
+)
+
+type CheckoutCmd struct {
+	v1alpha1.GitSource
+}
+
+func NewCheckoutCmd(s v1alpha1.GitSource) CheckoutCmd {
+	return CheckoutCmd{GitSource: s}
+}
+
+func (c CheckoutCmd) String() (string, error) {
+	var checkoutCommand string
+	var repository = c.Repository
+	var directory = c.Directory
+	var branch = c.Ref.Branch
+	var commit = c.Ref.Commit
+	var tag = c.Ref.Tag
+
+	repositoryName, err := c.ValidateRepo()
+	if err != nil {
+		return "", err
+	}
+
+	if directory == "" {
+		directory = defaultDirectory
+	}
+
+	if branch == "" {
+		branch = defaultBranch
+	}
+
+	if commit != "" {
+		checkoutCommand = fmt.Sprintf("git clone %s && cd %s && git checkout %s && cp -r %s/* /manifests",
+			repository, repositoryName, commit, directory)
+		return checkoutCommand, nil
+	}
+
+	if tag != "" {
+		checkoutCommand = fmt.Sprintf("git clone --depth 1 --branch %s %s && cd %s && git checkout tags/%s && cp -r %s/* /manifests",
+			tag, repository, repositoryName, tag, directory)
+		return checkoutCommand, nil
+	}
+
+	checkoutCommand = fmt.Sprintf("git clone --depth 1 --branch %s %s && cd %s && git checkout %s && cp -r %s/* /manifests",
+		branch, repository, repositoryName, branch, directory)
+	return checkoutCommand, nil
+}
+
+func (c CheckoutCmd) ValidateRepo() (string, error) {
+	u, err := url.Parse(c.Repository)
+	if err != nil {
+		return "", fmt.Errorf("provided git repository url malformed: %w", err)
+	}
```

Comment:
Does this work for the repo syntax that looks like this: `git@github.com:operator-framework/combo.git`?

Reply:
I feel pretty strongly we should just hand the repo string off to git as is and let git return any error it finds.

Reply:
This is largely to safely extract the name of the repository. If we have the ability to clone to a static repo name, we can remove this logic entirely.